pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/RangePredicateEvaluatorFactory.java

```diff
@@ -214,6 +219,16 @@ public int[] getMatchingDictIds() {
     public int getNumMatchingItems() {
       return Math.max(_numMatchingDictIds, 0);
     }
+
+    @Override
+    public int getInclusiveLowerBound() {
+      return getStartDictId();
+    }
+
+    @Override
+    public int getInclusiveUpperBound() {
+      return getEndDictId() - 1;
```

Comment:
Pay attention to how details like the last dictionary id being exclusive can be encapsulated with simple abstractions, rather than copying this snippet everywhere a range query needs to be evaluated against a dictionary. It's outside of the scope of this PR, but this pattern can be adopted in `SortedIndexBasedFilterOperator` to reduce coupling there, and would simplify the code changes required to introduce a sorted index without a dictionary in the future.